packages/proxy/src/providers/anthropic.ts

```diff
@@ -176,19 +179,43 @@ export interface AnthropicCompletion {
   model: string;
   stop_reason: string;
   stop_sequence: string | null;
-  usage: { input_tokens: number; output_tokens: number };
+  usage: {
+    input_tokens: number;
+    output_tokens: number;
+    cache_creation_input_tokens?: number;
+    cache_read_input_tokens?: number;
+  };
 }
 
 function updateUsage(
   anthropic: z.infer<typeof anthropicUsage>,
   openai: Partial<CompletionUsage>,
 ) {
   if (!isEmpty(anthropic.input_tokens)) {
-    openai.prompt_tokens = anthropic.input_tokens;
+    // OpenAI's convention is to accumulate all input tokens, including cached tokens.
+    openai.prompt_tokens =
```

Comment:
#nit would be nice to convert this to a pure function

Reply:
hm it's intentionally not -- the point is to accumulate metrics across streamed events